Add confirmation prompts and examples for Remove- functions

themilanfan opened the pull request:
Fixes #171

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/microsoft/PowerShellForGitHub?pullRequest=174) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/PowerShellForGitHub?pullRequest=174) before we can accept your contribution.<br/><br/>:x: themilanfan [sign now](https://cla.opensource.microsoft.com/microsoft/PowerShellForGitHub?pullRequest=174)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/microsoft/PowerShellForGitHub?pullRequest=174) it.</sub>

HowardWolosky commented on GitHubLabels.ps1:
```diff
@@ -858,12 +862,16 @@ function Remove-GitHubIssueLabel
         Removes the label called "TestLabel" from issue 1 in the PowerShellForGitHub project.
+
+    .EXAMPLE
+        Remove-GitHubIssueLabel -OwnerName Microsoft -RepositoryName PowerShellForGitHub -Name TestLabel -Issue 1
```
You didn't actually specify `-Confim:$false` ;)

```suggestion
        Remove-GitHubIssueLabel -OwnerName Microsoft -RepositoryName PowerShellForGitHub -Name TestLabel -Issue 1 -Confirm:$false
```

HowardWolosky commented on GitHubAssignees.ps1:
```diff
@@ -323,12 +323,16 @@ function Remove-GithubAssignee
 
     .EXAMPLE
         Remove-GithubAssignee -OwnerName Microsoft -RepositoryName PowerShellForGitHub -Assignee $assignees
-
```
Here and elsewhere: The rest of this module typically keeps a linefeed between the code and the example description.  A couple exceptions managed to creep in over time, but I'd prefer we keep the separation between the code and the explanation.

themilanfan commented:
@HowardWolosky thank you for the feedback, I've just committed the changes!

themilanfan commented:
@HowardWolosky not sure what's happening with the check, I reviewed all the changes you requesteed

HowardWolosky commented:
@themilanfan - Thanks for the update.  Will review the changes this weekend.

As for the checks, ignore them for now.  I've been running into issues with the tests properly running in PR CI builds, so I still need to verify them locally before merging.  I'll get to fixing the PR CI build at some point.

HowardWolosky commented on GitHubRepositories.ps1:
```diff
@@ -217,10 +217,16 @@ function Remove-GitHubRepository
+    .EXAMPLE
+        Remove-GitHubRepository -Uri https://github.com/You/YourRepoToDelete -Confirm:$false
+
+        Remove repository given uri without prompting for confirmation.
```
```suggestion
        Remove repository with the given URI, without prompting for confirmation.
```

themilanfan commented:
Hi @HowardWolosky, I've added the remaining -Confirm:$false as well as updated the documentation and fixed another small issue in GitHubLabels.tests.ps1 as well. I will also wait for this to merge so that for issue #176 I don't miss any Confirm:$false there.

HowardWolosky commented on GitHubAssignees.ps1:
```diff
@@ -328,10 +328,16 @@ function Remove-GithubAssignee
-        DefaultParameterSetName='Elements')]
+        DefaultParameterSetName='Elements',
+        ConfirmImpact="High")]
     [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSShouldProcess", "", Justification="Methods called within here make use of PSShouldProcess, and the switch is passed on to them inherently.")]
```
```suggestion
```

For this one and the others, we no longer need this exception, because we have an explicit usage of `$PSCmdlet.ShouldProcess` directly in the method now.

themilanfan commented:
Remove them 😄

HowardWolosky commented:
It looks like some conflicts came in as a result of some of the recent commits as I've been trying to get the CI to run clean on all platforms.  Can you please address the conflicts?  Then I'll give it a run through the CI and get it committed....Thanks!

themilanfan commented:
Resolved conflicts 😄

HowardWolosky commented:
/azp run PowerShellForGitHub-CI
